MailAlert.php

```diff
@@ -170,14 +170,11 @@ public static function getMailAlerts($id_customer, $id_lang, Shop $shop = null)
     public static function sendCustomerAlert($id_product, $id_product_attribute)
     {
         $link = new Link();
-        $context = Context::getContext()->cloneContext();
-        $customers = self::getCustomers($id_product, $id_product_attribute);
+        $customers = self::getCustomers($id_product, $id_product_attribute, Context::getContext()->shop->id);
 
         foreach ($customers as $customer) {
             $id_shop = (int) $customer['id_shop'];
             $id_lang = (int) $customer['id_lang'];
-            $context->shop->id = $id_shop;
```

Comment:
$context->shop->id is used in Product::getProductName() 3 lines below. If the product name differs from shop to shop, the wrong product name might be sent to the customer.

Reply:
Fixed